### ISSUE TITLE
PWA Offline Caching Issue with Module JS URL's including the version number.

### ISSUE THREAD

**Issue by WayneHiller** [Megabit/Blazorise#4594]
Blazorise Boorstrap5 1.2.

I have been checking my PWA Blazor Wasm app's operation while offline.  
When publishing, dotnet records all the offline assets to be cached in service-worker-assets.js such as:
```
...
{
      "hash": "sha256-hVhIZcaWf6qmi4VdItR\/L+gH+5bL3bJAevfwNwLN5ks=",
      "url": "_content\/Blazorise\/inputMask.js"
    },
    {
      "hash": "sha256-mbKL6XwKZyyrts0POi77EYKAsiuPcX8ZEDw5bwtQtFY=",
      "url": "_content\/Blazorise\/io.js"
    },
    {
      "hash": "sha256-EJyX6lPGUR\/sv5GUBR6t8lXwows9MNteaCtZ0YU\/kv4=",
      "url": "_content\/Blazorise\/memoEdit.js"
    },
    {
      "hash": "sha256-RWfYgydRf807XC5rZiHoGBMxhaCyipvxqAi3rtopbGo=",
      "url": "_content\/Blazorise\/numericPicker.js"
    },
...
```

Now the problem is that these are URL's without the ?v=1.2.0.0
And the Module system in Blazorise always appends the version param when dynamically loading the JS files.

Is there a way that Blazorize could be changed to include a global setting that could be set to not append the version to the asset URL?

**Comment by stsrki** [comment]
Making the version number optional would be easy on the C# side. But the problem is that we use the hardcoded version in the JavaScipt also to import JS modules. And JavaScript doesn't allow for non-hardcoded JS modules file paths.

eg. https://github.com/Megabit/Blazorise/blob/master/Source/Blazorise/wwwroot/button.js#L1

---

Can you modify `service-worker-assets.js` to include version numbers?

**Comment by David-Moreira** [comment]
@WayneHiller is this actually a problem? Doesn't the query string get ignored if not found, and it still loads the files successfully?

**Comment by WayneHiller** [comment]
> @WayneHiller is this actually a problem? Doesn't the query string get ignored if not found, and it still loads the files successfully?

The problem is that with the version param it does not find it in the cache so it tries to get it over the network and it is offline so the call fails.


> Making the version number optional would be easy on the C# side. But the problem is that we use the hardcoded version in the JavaScipt also to import JS modules. And JavaScript doesn't allow for non-hardcoded JS modules file paths.
> 
> eg. https://github.com/Megabit/Blazorise/blob/master/Source/Blazorise/wwwroot/button.js#L1
> 
> Can you modify `service-worker-assets.js` to include version numbers?

Oh ya I forgot about that part. I will build an app that adds the version to the URL's in the service-worker-assets.js file after publish.

**Comment by stsrki** [comment]
Just a helpful note. We always use the full version with four digits, eg, `1.2.0.0`

**Comment by stsrki** [comment]
@WayneHiller Did you try updating `service-worker-assets.js`?

**Comment by WayneHiller** [comment]
@stsrki Yes, I ended up creating a little console app that does the publish process and then updates the service-worker-assets.js file at the end. There was other things I needed to do anyway. Works fine.

**Comment by stsrki** [comment]
In that case, I believe it is safe to close the ticket.
